torch_xla/experimental/xla_sharded_tensor.py

```diff
@@ -32,6 +32,76 @@ class XLAShardedTensor(torch.Tensor):
     to the graph before compilation.
   """
 
+  # XLAShardedTensor behaves like a unpartitioned,
+  # combined tensor on the host machine. When user annotates,
+  # this is simply set to the input tensor. When an XLA partitioned
+  # output tensor returns (or sharding propagated intermediate tensors)
+  # as XLAShardedTensor, the backend gathers global data across devices
+  # and materialize and set `global_tensor` on the host; the actual device
+  # data still remain on individual device as sharded or replicated.
+  # Note: we should drop this reference, and force all gather on each access.
+  global_tensor: torch.Tensor
```

Comment:
Are these fields and this class tested? It seems pretty random in this pr.

Reply:
It's part of the first separate PR -- once we land it, rebase and this will be gone.

Reply:
@JackCaoG correction, this is the first PR. And yes, this is random and wrong -- the same code is copied over twice lol Everything, except "Not implemented" methods are tested -- once we land this, Steven will add more extensive unit tests around, too.